File: shared/src/main/Plume/Types/ServerTypes.kt
package Plume.Types

import kotlinx.serialization.Serializable

@Serializable
data class AllocationData(
    val ip: String,
    val port: Int,
)

data class StartCore(
    val installScript: String = "",
    val installImage: String = "",
    val installEntrypoint: String = "",

    val startupCommand: String = "",
    val startupScript: String = "",
    val dockerEntrypoint: String = "",

    val stopCommand: String = "",

    val configSystem: Any? = null,
    val startupParser: Any? = null,

    val rootAcess: Long? = null,
    val maintainable: Long? = null,
)

data class StartData(
    var image: String,
    val memory: Int,
    val cpu: Int,
    val environment: Any? = null,

    val primaryAllocation: AllocationData? = null,
    val additionalAllocation: List<AllocationData> = emptyList(),

    val disk: Long = 0,
    val core: StartCore,
)

data class UsageMetrics(
    val cpu: Double = 0.0,
    val memory: Long = 0,
    val memoryLimit: Long = 0,
    val disk: Long = 0,
    val networkIn: Long = 0,
    val networkOut: Long = 0,
)

File: shared/src/Plume/Types/ServerTypes.cs


namespace Plume.Types;


// Equivalente ao '@Serializable data class AllocationData'
public record AllocationData(string Ip, int Port);


// Equivalente ao 'data class StartCore'
public record StartCore {
  public string InstallScript {
    get;
    init;
  } = "";
  public string InstallImage {
    get;
    init;
  } = "";
  public string InstallEntrypoint {
    get;
    init;
  } = "";
  public string StartupCommand {
    get;
    init;
  } = "";
  public string StartupScript {
    get;
    init;
  } = "";
  public string DockerEntrypoint {
    get;
    init;
  } = "";
  public string StopCommand {
    get;
    init;
  } = "";
  // Kotlin 'Any?' mapeia perfeitamente para 'object?' no C#
  public object ? ConfigSystem {
    get;
    init;
  } = null;
  public object ? StartupParser {
    get;
    init;
  } = null;
  // Mantida a grafia original do Kotlin (rootAcess)
  public long ? RootAcess {
    get;
    init;
  } = null;
  public long ? Maintainable {
    get;
    init;
  } = null;
}
// Equivalente ao 'data class StartData'
public record StartData {
  // Como no Kotlin era 'var', em C# permitimos escrita com 'get; set;'
  public string Image {
    get;
    set;
  } = "";
  public int Memory {
    get;
    init;
  }
  public int Cpu {
    get;
    init;
  }
  public object ? Environment {
    get;
    init;
  } = null;
  public AllocationData ? PrimaryAllocation {
    get;
    init;
  } = null;
  public List < AllocationData > AdditionalAllocation {
    get;
    init;
  } = new List < AllocationData > ();
  public long Disk {
    get;
    init;
  } = 0;
  public StartCore Core {
    get;
    init;
  } = null!;
}
// Equivalente ao 'data class UsageMetrics'
public record UsageMetrics {
  public double Cpu {
    get;
    init;
  } = 0.0;
  public long Memory {
    get;
    init;
  } = 0;
  public long MemoryLimit {
    get;
    init;
  } = 0;
  public long Disk {
    get;
    init;
  } = 0;
  public long NetworkIn {
    get;
    init;
  } = 0;
  public long NetworkOut {
    get;
    init;
  } = 0;
}